test/batch_test.rb

```diff
@@ -9,10 +9,13 @@ def test_batch
   end
 
   def test_nested_batch
-    GraphQL::Batch.batch do
-      assert_raises(GraphQL::Batch::NestedError) do
-        GraphQL::Batch.batch {}
+    product = GraphQL::Batch.batch do
+      promise = RecordLoader.for(Product).load(1)
+      GraphQL::Batch.batch do
+        RecordLoader.for(Product).load(1)
       end
+      promise
     end
+    assert_equal 'Shirt', product.title
```

Comment:
Maybe also a `assert_nil GraphQL::Batch::Executor.current` check to make sure the cleanup code works on the outermost `GraphQL::Batch.batch`

Reply:
Done.